flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/CheckpointConfig.java

```diff
@@ -52,6 +52,15 @@
 @Public
 public class CheckpointConfig implements java.io.Serializable {
 
+    // NOTE TO IMPLEMENTERS:
+    // Please do not add further fields to this class. Use the ConfigOption stack instead!
+    // It is currently very tricky to keep this kind of POJO classes in sync with instances of
+    // org.apache.flink.configuration.Configuration. Instances of Configuration are way easier to
+    // pass, layer, merge, restrict, copy, filter, etc.
+    // See ExecutionOptions.RUNTIME_MODE for a reference implementation. If the option is very
+    // crucial for the API, we can add a dedicated setter to StreamExecutionEnvironment. Otherwise,
+    // introducing a ConfigOption should be enough.
```

Comment:
Should we rather make an effort to deprecate these? (same applies to ExecutionConfig)

Reply:
Impossible right now. The impact is huge. I just try to stop people from making the impact even worse. Everyone that worked with configuration in SDK teams know that we need to change something here, but there were no resources for this yet.

Reply:
Created https://issues.apache.org/jira/browse/FLINK-29379